starlette/middleware/base.py

```diff
@@ -28,12 +28,22 @@ async def call_next(request: Request) -> Response:
             app_exc: typing.Optional[Exception] = None
             send_stream, recv_stream = anyio.create_memory_object_stream()
 
+            async def send(msg: Message) -> None:
+                # Shield send "http.response.start" from cancellation.
+                # Otherwise, `await recv_stream.receive()` will raise `anyio.EndOfStream` if request is disconnected,
+                # due to `task_group.cancel_scope.cancel()` in `StreamingResponse.__call__.<locals>.wrap`
+                # and cancellation check in `await checkpoint()` of `MemoryObjectSendStream.send`,
+                # and then `RuntimeError: No response returned.` will be raised below.
```

Comment:
thank you for the comment explaining what's going on! I shortened the lines to fix listing and did some minor rewording

```suggestion
                # Shield send "http.response.start" from cancellation.
                # Otherwise, `await recv_stream.receive()` will raise `anyio.EndOfStream`
                # if the connection is disconnected,
                # due to `task_group.cancel_scope.cancel()` in
                # `StreamingResponse.__call__.<locals>.wrap`
                # and cancellation check in `await checkpoint()` of
                # `MemoryObjectSendStream.send`
                # This triggers the check we have in this middleware resulting in
                # `RuntimeError: No response returned.` will be raised below.
```